### ISSUE TITLE
[bug] fractional_max_pool2d backward: CUDA doesn't support input without batch dimenion, CPU does

### ISSUE THREAD

**Issue by kshitij12345** [pytorch/pytorch#52427]
```python
import torch
from torch.autograd import gradcheck

torch.set_default_dtype(torch.double)

device='cuda'

x = torch.randn(2, 7, 7, requires_grad=True, device=device)
samples = x.new(1, 2, 2).uniform_()

def fn(x):
    return torch.nn.functional.fractional_max_pool2d(
                x, (2, 2), output_size=(3, 3), _random_samples=samples)

gradcheck(fn, [x])
```

Above script works with `device='cpu'`, while with `'cuda'`, it fails with 
`RuntimeError: TensorAccessor expected 4 dims but tensor has 3` in gradcheck.

cc @albanD @mruberry @jbschlosser @heitorschueroff

**Comment by jbschlosser** [comment]
Confirmed that the CUDA forward pass works without the batch dim, but CUDA backward pass does not.

**Comment by khushi-411** [comment]
Hi,
I think this issue is fixed in the master branch (`'1.13.0a0+git016fcca'`). Please see:
```python
In [3]: torch.set_default_dtype(torch.double)

In [4]: device='cuda'

In [5]: x = torch.randn(2, 7, 7, requires_grad=True, device=device)
   ...: samples = x.new(1, 2, 2).uniform_()

In [6]: def fn(x):
   ...:     return torch.nn.functional.fractional_max_pool2d(
   ...:                 x, (2, 2), output_size=(3, 3), _random_samples=samples)
   ...: 

In [7]: gradcheck(fn, [x])
   ...: 
Out[7]: True
```
This issue can be closed as completed. Could you please re-check it? Thanks!!

**Comment by albanD** [comment]
This is indeed fixed in 1.12.1
